File: backend/src/main/java/com/daedan/festabook/lostitem/dto/LostItemResponse.java
package com.daedan.festabook.lostitem.dto;

import com.daedan.festabook.lostitem.domain.LostItem;
import com.daedan.festabook.lostitem.domain.PickupStatus;
import java.time.LocalDateTime;

public record LostItemResponse(
        Long lostItemId,
        String imageUrl,
        String storageLocation,
        PickupStatus status,
        LocalDateTime createdAt
) {

    public static LostItemResponse from(LostItem lostItem) {
        return new LostItemResponse(
                lostItem.getId(),
                lostItem.getImageUrl(),
                lostItem.getStorageLocation(),
                lostItem.getStatus(),
                lostItem.getCreatedAt()
        );
    }
}

File: android/app/src/main/java/com/daedan/festabook/data/model/response/lostitem/LostItemResponse.kt
package com.daedan.festabook.data.model.response.lostitem

import com.daedan.festabook.domain.model.Lost
import com.daedan.festabook.domain.model.LostItemStatus
import com.daedan.festabook.domain.model.toLocalDateTime
import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable
import timber.log.Timber

@Serializable
data class LostItemResponse(
    @SerialName("lostItemId")
    val lostItemId: Long,
    @SerialName("imageUrl")
    val imageUrl: String,
    @SerialName("storageLocation")
    val storageLocation: String,
    @Serializable(with = LostItemStatusSerializer::class)
    @SerialName("pickupStatus")
    val pickupStatus: LostItemStatus,
    @SerialName("createdAt")
    val createdAt: String,
)

fun LostItemResponse.toDomain(): Lost.Item =
    Lost.Item(
        lostItemId = lostItemId,
        imageUrl = imageUrl,
        storageLocation = storageLocation,
        status =
            pickupStatus.also { if (it == LostItemStatus.UNKNOWN) Timber.e("${::toDomain.name}: 알 수 없는 분실물 상태") },
        createdAt = createdAt.toLocalDateTime(),
    )
